File: src/main/java/me/foodbag/hello/persistence/repository/FoodRepository.java
package me.foodbag.hello.persistence.repository;

import me.foodbag.hello.persistence.model.Food;
import org.springframework.data.repository.CrudRepository;
import java.util.List;

public interface FoodRepository extends CrudRepository<Food, Long> {

  Food findByName(String name);

  @Override
  List<Food> findAll();

}

File: src/main/java/me/foodbag/hello/persistence/repository/FoodRepository.kt
package me.foodbag.hello.persistence.repository

import me.foodbag.hello.persistence.model.Food
import org.springframework.data.repository.CrudRepository

interface FoodRepository : CrudRepository<Food, Long> {

    fun findByName(name: String): Food

    override fun findAll(): List<Food>

}
